examples/all-clusters-app/linux/WindowCoveringManager.cpp

```diff
@@ -49,52 +49,86 @@ void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aA
     uint16_t currentPosition = manager->mCurrentPosition.Value();
     uint16_t targetPosition  = manager->mTargetPosition.Value();
 
-    ChipLogProgress(NotSpecified, "HandleMovementTimer:currentPosition:%u, targetPosition:%u", currentPosition, targetPosition);
+    ChipLogProgress(NotSpecified, "HandleLiftMovementTimer:currentPosition:%u, targetPosition:%u", currentPosition, targetPosition);
 
     if (OperationalState::MovingUpOrOpen == manager->mState)
     {
-        if (currentPosition > targetPosition)
-        {
-            uint16_t tempPosition =
-                currentPosition > kDefaultMovementStep ? static_cast<uint16_t>(currentPosition - kDefaultMovementStep) : 0;
-            currentPosition = tempPosition > targetPosition ? tempPosition : targetPosition;
-
-            manager->mCurrentPosition.SetNonNull(currentPosition);
-        }
-        else
+        if (ComputeOperationalState(targetPosition, currentPosition) != OperationalState::MovingUpOrOpen)
         {
             ChipLogProgress(NotSpecified, "Reached the target position");
             return;
         }
+
+        uint16_t tempPosition =
+            currentPosition > kDefaultMovementStep ? static_cast<uint16_t>(currentPosition - kDefaultMovementStep) : 0;
+        currentPosition = tempPosition > targetPosition ? tempPosition : targetPosition;
+
+        manager->mCurrentPosition.SetNonNull(currentPosition);
     }
     else
     {
-        if (currentPosition < targetPosition)
-        {
-            uint16_t tempPosition = static_cast<uint16_t>(currentPosition + kDefaultMovementStep);
-            currentPosition       = tempPosition < targetPosition ? tempPosition : targetPosition;
-            manager->mCurrentPosition.SetNonNull(currentPosition);
-        }
-        else
+        if (ComputeOperationalState(targetPosition, currentPosition) != OperationalState::MovingDownOrClose)
         {
             ChipLogProgress(NotSpecified, "Reached the target position");
             return;
         }
+
+        uint16_t tempPosition = static_cast<uint16_t>(currentPosition + kDefaultMovementStep);
```

Comment:
please use this function for delta step
Percent100ths ComputePercent100thsStep(OperationalState direction, Percent100ths previous, Percent100ths delta);